tests/web_api/detect_test.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import unittest
+import importlib
+utils = importlib.import_module(
+    'extensions.sd-webui-controlnet.tests.utils', 'utils')
+utils.setup_test_env()
+
+
+class TestDetectEndpointWorking(unittest.TestCase):
+    def setUp(self):
+        self.base_detect_args = {
+            "controlnet_module": "canny",
+            "controlnet_input_images": [utils.readImage("test/test_files/img2img_basic.png")],
+            "controlnet_processor_res": 512,
+            "controlnet_threshold_a": 0,
+            "controlnet_threshold_b": 0,
+        }
+
+    def test_detect_with_invalid_module_performed(self):
+        detect_args = self.base_detect_args.update({
+            "controlnet_module": "INVALID",
+        })
+        self.assertEqual(utils.detect(detect_args).status_code, 500)
+
+    def test_detect_with_no_input_images_performed(self):
+        detect_args = self.base_detect_args.update({
```

Comment:
update returns `None` and updates the dict in place. Maybe just use `self.base_detect_args` directly in the assert below instead.

Reply:
Looks like my python is a bit rusty :)
I've switched it up to handle it without mutating the base args.